tests/ha/test_multi_server_pod_failure.py

```diff
@@ -629,3 +629,383 @@ def test_writes_till_kserver_pods_fail(self):
+        counter = 0
+        for pod_name in self.pod_name_list:
+            hostname = self.pod_dict.get(pod_name)[0]
+            resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=[pod_name], fail=True,
+                                                               hostname=hostname)
+            LOGGER.debug("Services status on %s : %s", pod_name, resp)
+            if not resp[0]:
+                counter += 1
+            server_data_pod_list.remove(pod_name)
+        assert_utils.assert_equal(counter, 0, "Services on some server pods not stopped.")
+        LOGGER.info("Step 4: Services of server pods are in offline state")
+
+        LOGGER.info("Step 5: Check services status on remaining pods %s", server_data_pod_list)
+        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=server_data_pod_list,
+                                                           fail=False)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 5: Services of remaining pods are in online state")
+
+        LOGGER.info("Step 6: Perform WRITEs-READs-Verify and verify DI on the written data")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix,
+                                                    skipcleanup=True, nsamples=2, nclients=2)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 6: Performed WRITEs-READs-Verify and verified DI on the written data")
+
+        LOGGER.info("Step 7: Perform WRITEs-READs-Verify with variable object sizes. 0B + ("
+                    "1KB - 512MB) ")
+        users = self.mgnt_ops.create_account_users(nusers=1)
+        self.test_prefix = 'test-40579-1'
+        self.s3_clean.update(users)
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix,
+                                                    skipcleanup=True, nsamples=2, nclients=2)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 7: Performed WRITEs-READs-Verify with variable sizes objects ")
+
+        LOGGER.info("Completed: Test to verify degraded WRITEs after all K server pods "
+                    "down - unsafe shutdown.")
+
+    # pylint: disable=too-many-statements
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-40580")
+    @CTFailOn(error_handler)
+    def test_deletes_till_kserver_pods_fail(self):
+        """
+        Test to verify degraded DELETEs after each pod failure till K server pods
+        down - unsafe shutdown.
+        """
+        LOGGER.info("Started: Test to verify degraded Deletes after each pod failure till "
+                    "K server pods down - unsafe shutdown.")
+        wr_bucket = HA_CFG["s3_bucket_data"]["no_buckets_for_deg_deletes"]
```

Comment:
we need to change the logic here, we can't keep delete 20 buckets from count of 150 because what if K values goes more for 15 Nodes. 
Please take a referrence from test_partial_mpu_after_kpods_fail which has updated 
total_parts = self.kvalue * 5 + HA_CFG["5gb_mpu_data"]["total_parts"] according to the K value need. We can update the same here
@ApurwaMohite @sonalk0209  Please suggest your thoughts